CMakeLists.txt

```diff
@@ -76,26 +78,27 @@ add_executable(test_pso
 target_include_directories(test_pso PRIVATE ${CRITERION_INCLUDE_DIRS})
 target_link_libraries(test_pso PRIVATE ${CRITERION_LIBRARIES})
 
-##### pso rbf integration test executable  ######
-#add_executable(test_integration_psorbf
-#        tests/test_integration_pso_rbf.c
-#        src/pso_rbf.c
-#        src/objectives.c
-#        src/utils.c)
-#target_include_directories(test_integration_psorbf PRIVATE ${CRITERION_INCLUDE_DIRS})
-#target_link_libraries(test_integration_psorbf
-#        PRIVATE ${CRITERION_LIBRARIES}
-#        m)
-#
-##### pso rbf unit test ######
-#add_executable(test_psorbf
-#        tests/test_pso_rbf.c
-#        src/cpp_utils.cpp
-#        src/pso_rbf.c
-#        src/utils.c
-#        src/objectives.c)
-#target_include_directories(test_psorbf PRIVATE ${CRITERION_INCLUDE_DIRS})
-#target_link_libraries(test_psorbf PRIVATE ${CRITERION_LIBRARIES})
+##### squirrel integration test executable  ######
+add_executable(test_integration_squirrel
+        tests/test_integration_squirrel.c
+        src/squirrel.c
+        src/objectives.c
+        src/utils.c)
+target_include_directories(test_integration_squirrel PRIVATE ${CRITERION_INCLUDE_DIRS})
+target_link_libraries(test_integration_squirrel
+        PRIVATE ${CRITERION_LIBRARIES}
+        m)
+
+##### squirrel unit test ######
+add_executable(test_squirrel
+        tests/test_squirrel.c
+        src/cpp_utils.cpp
+        src/squirrel.c
+        src/utils.c
+        src/objectives.c)
+target_include_directories(test_squirrel PRIVATE ${CRITERION_INCLUDE_DIRS})
+target_link_libraries(test_squirrel PRIVATE ${CRITERION_LIBRARIES})
+
 
 ##### Penguin integration test executable ######

```

Comment:
Please add your unit tests to `test_unit` as well. And add the integration tests as a separate test with:

    add_test(integration_squirrel
      test_integration_squirrel -q)